src/essreflectometry/amor/load.py

```diff
@@ -17,10 +17,9 @@
 
 
 def load_detector(
-    filepath: FilePath[Run], detector: NeXusDetectorPath[Run]
+    file_path: FilePath[Run], detector_name: NeXusDetectorName[Run]
 ) -> RawDetector[Run]:
-    with snx.File(filepath, 'r') as f:
-        return RawDetector[Run](f[detector][()])
+    return nexus.load_detector(file_path=file_path, detector_name=detector_name)
 
 
 def load_events(detector: RawDetector[Run]) -> RawEvents[Run]:
```

Comment:
This function should use https://scipp.github.io/essreduce/generated/modules/ess.reduce.nexus.extract_detector_data.html instead of relying on the field name.